### ISSUE TITLE
Errors while making Drakvuf for Debugging

### ISSUE THREAD

**Issue by ubersandro** [tklengyel/drakvuf#1629]
I experienced some compilation errors while MAKING Drakvuf for my Intell11th gen powered AIO machine. In particular, I experienced some C++ implicit casts errors while making using the default makefile.  Did you experience them as well? Is it safe to suppress this kind of error by removing a particular compilation flag? If yes, which flag can I remove?

**Comment by ubersandro** [comment]
I am making on Ubuntu Focal Fossa 20.4.6.

**Comment by tklengyel** [comment]
We regularly test the build process on [Ubuntu 20.04](https://github.com/tklengyel/drakvuf/actions/runs/4547845742/jobs/8018234614) and we haven't seen any errors on the CI. Might be more helpful if you actually posted the errors you are seeing.

**Comment by ubersandro** [comment]
Good morning and thanks for you fast answer. Here is the stderr resulting from the making of drakvuf configured for debug. 
-------------------------------------------------------------------------------------------------------------------
xen_helper.c: In function ‘xen_init_interface’:
xen_helper.c:146:44: error: request for implicit conversion from ‘void *’ to ‘xentoollog_logger_stdiostream * (*)(FILE *, xentoollog_level,  unsigned int)’ {aka ‘struct xentoollog_logger_stdiostream * (*)(struct _IO_FILE *, enum xentoollog_level,  unsigned int)’} not permitted in C++ [-Werror=c++-compat]
  146 |     _xen->xlw.xtl_createlogger_stdiostream = dlsym(_xen->xlw.xtl_handle, "xtl_createlogger_stdiostream");
      |                                            ^
xen_helper.c:147:34: error: request for implicit conversion from ‘void *’ to ‘void (*)(struct xentoollog_logger *)’ not permitted in C++ [-Werror=c++-compat]
  147 |     _xen->xlw.xtl_logger_destroy = dlsym(_xen->xlw.xtl_handle, "xtl_logger_destroy");
      |                                  ^
xen_helper.c:158:31: error: request for implicit conversion from ‘void *’ to ‘int (*)(libxl_ctx **, int,  unsigned int,  xentoollog_logger *)’ {aka ‘int (*)(struct libxl__ctx **, int,  unsigned int,  struct xentoollog_logger *)’} not permitted in C++ [-Werror=c++-compat]
  158 |     _xen->xlw.libxl_ctx_alloc = dlsym(_xen->xlw.xl_handle, "libxl_ctx_alloc");
      |                               ^
xen_helper.c:159:30: error: request for implicit conversion from ‘void *’ to ‘int (*)(libxl_ctx *)’ {aka ‘int (*)(struct libxl__ctx *)’} not permitted in C++ [-Werror=c++-compat]
  159 |     _xen->xlw.libxl_ctx_free = dlsym(_xen->xlw.xl_handle, "libxl_ctx_free");
      |                              ^
xen_helper.c:160:35: error: request for implicit conversion from ‘void *’ to ‘int (*)(libxl_ctx *, const char *, uint32_t *)’ {aka ‘int (*)(struct libxl__ctx *, const char *, unsigned int *)’} not permitted in C++ [-Werror=c++-compat]
  160 |     _xen->xlw.libxl_name_to_domid = dlsym(_xen->xlw.xl_handle, "libxl_name_to_domid");
      |                                   ^
xen_helper.c:161:35: error: request for implicit conversion from ‘void *’ to ‘char * (*)(libxl_ctx *, uint32_t)’ {aka ‘char * (*)(struct libxl__ctx *, unsigned int)’} not permitted in C++ [-Werror=c++-compat]
  161 |     _xen->xlw.libxl_domid_to_name = dlsym(_xen->xlw.xl_handle, "libxl_domid_to_name");
      |                                   ^
xen_helper.c:162:42: error: request for implicit conversion from ‘void *’ to ‘int (*)(libxl_ctx *, uint32_t,  const char *, char **, const libxl_asyncop_how *)’ {aka ‘int (*)(struct libxl__ctx *, unsigned int,  const char *, char **, const struct <anonymous> *)’} not permitted in C++ [-Werror=c++-compat]
  162 |     _xen->xlw.libxl_qemu_monitor_command = dlsym(_xen->xlw.xl_handle, "libxl_qemu_monitor_command");
      |                                          ^
xen_helper.c:183:37: error: request for implicit conversion from ‘void *’ to ‘xenforeignmemory_handle * (*)(struct xentoollog_logger *, unsigned int)’ {aka ‘struct xenforeignmemory_handle * (*)(struct xentoollog_logger *, unsigned int)’} not permitted in C++ [-Werror=c++-compat]
  183 |     _xen->xlw.xenforeignmemory_open = dlsym(_xen->xlw.xfm_handle, "xenforeignmemory_open");
      |                                     ^
xen_helper.c:184:38: error: request for implicit conversion from ‘void *’ to ‘int (*)(xenforeignmemory_handle *)’ {aka ‘int (*)(struct xenforeignmemory_handle *)’} not permitted in C++ [-Werror=c++-compat]
  184 |     _xen->xlw.xenforeignmemory_close = dlsym(_xen->xlw.xfm_handle, "xenforeignmemory_close");
      |                                      ^
xen_helper.c:185:46: error: request for implicit conversion from ‘void *’ to ‘int (*)(xenforeignmemory_handle *, domid_t,  unsigned int,  unsigned int,  size_t *)’ {aka ‘int (*)(struct xenforeignmemory_handle *, short unsigned int,  unsigned int,  unsigned int,  long unsigned int *)’} not permitted in C++ [-Werror=c++-compat]
  185 |     _xen->xlw.xenforeignmemory_resource_size = dlsym(_xen->xlw.xfm_handle, "xenforeignmemory_resource_size");
      |                                              ^
xen_helper.c:186:45: error: request for implicit conversion from ‘void *’ to ‘xenforeignmemory_resource_handle * (*)(xenforeignmemory_handle *, domid_t,  unsigned int,  unsigned int,  long unsigned int,  long unsigned int,  void **, int,  int)’ {aka ‘struct xenforeignmemory_resource_handle * (*)(struct xenforeignmemory_handle *, short unsigned int,  unsigned int,  unsigned int,  long unsigned int,  long unsigned int,  void **, int,  int)’} not permitted in C++ [-Werror=c++-compat]
  186 |     _xen->xlw.xenforeignmemory_map_resource = dlsym(_xen->xlw.xfm_handle, "xenforeignmemory_map_resource");
      |                                             ^
xen_helper.c:187:47: error: request for implicit conversion from ‘void *’ to ‘int (*)(xenforeignmemory_handle *, xenforeignmemory_resource_handle *)’ {aka ‘int (*)(struct xenforeignmemory_handle *, struct xenforeignmemory_resource_handle *)’} not permitted in C++ [-Werror=c++-compat]
  187 |     _xen->xlw.xenforeignmemory_unmap_resource = dlsym(_xen->xlw.xfm_handle, "xenforeignmemory_unmap_resource");
      |                                               ^
cc1: all warnings being treated as errors
make[3]: *** [Makefile:542: xen_helper.lo] Error 1
make[2]: *** [Makefile:776: all-recursive] Error 1
make[1]: *** [Makefile:543: all-recursive] Error 1
make: *** [Makefile:475: all] Error 2
-------------------------------------------------------------------------------------------------------------------------

**Comment by tklengyel** [comment]
What compiler and configuration are you using to trigger this error?

**Comment by ubersandro** [comment]
I used  the compiler gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0 and the makefile is the one resulting from following as it is the guide on the official Drakvuf site for building everything up to Drakvuf (this latter excluded, of course). For building Drakvuf I followed the guide in the Drakvuf wiki which explains how to make the DEBUG version. In the end, the makefile is the one resulting from ./configure --enable-debug.
